### PR TITLE
Piper/add some basic tests

### PR DESCRIPTION
**Pull request by pipermerriam**
builds on https://github.com/esPass/pretix-eth-payment-plugin/pull/21

### What was wrong?

No real tests

### How was it fixed?

Added some basic tests for the payment provider functionality

- tests that basic properties are set
- test that the form fields for settings are properly added
- test that the payment form fields are properly generated.

237BEFE100000578-2848550-Anna_says_her_photos_have_changed_peoples_opinions_on_rats_even_-2_1416923363929

### CODE REVIEW

File: pretix_eth/payment.py [hunk 1]
```diff
@@ -50,15 +50,17 @@ def settings_form_fields(self):
...
+            (self.settings.DAI or self.settings.ETH) and super().is_allowed(request)
+        )
 
     @property
     def payment_form_fields(self):
         e = ('ETH', _('Ethereum'))
         d = ('DAI', _('DAI'))
         if self.settings.ETH and self.settings.DAI:
-            tup = (e, d)
+            tup = (d, e)
```
Comment (carver) [inline pretix_eth/payment.py]: Why does order matter here? Maybe deserves a comment.

_Edit: Ah, maybe it affects the order it shows up in the payment form?_

File: tests/test_ethereum_payment_provider.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        presale_start=presale_start_at,
+        presale_end=presale_end_at,
+        location='Osaka',
+    )
+    return event
+
+
+@pytest.fixture
+def provider(event):
+    provider = Ethereum(event)
```
Comment (carver) [inline tests/test_ethereum_payment_provider.py]: `Ethereum` seems like a problematic name for a class, but seems out of scope for the PR. I suppose you could import `payment` so it would be `payment.Ethereum(event)` here, if you like that better.

File: tests/test_ethereum_payment_provider.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    return provider
+
+
+@pytest.mark.django_db
+def test_provider_properties(provider):
+    assert provider.identifier == 'ethereum'
+    assert provider.verbose_name == 'Ethereum'
+    assert provider.public_name == 'Ethereum'
```
Comment (carver) [inline tests/test_ethereum_payment_provider.py]: What is a `provider` in this context? Should this be "Ether" instead of "Ethereum"? Or maybe it should be the "Ethereum Foundation"? Or "Ethereum-based"?

File: tests/test_ethereum_payment_provider.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+@pytest.mark.django_db
+def test_provider_payment_form_fields_both_ETH_and_DAI(provider):
+    provider.settings.set('DAI', DAI_ADDRESS)
+    provider.settings.set('ETH', ETH_ADDRESS)
+
+    payment_form_fields = provider.payment_form_fields
+    currency_type_field = payment_form_fields['currency_type']
+    assert len(currency_type_field.choices) == 2
+    dai_choice = currency_type_field.choices[0]
+    eth_choice = currency_type_field.choices[1]
```
Comment (carver) [inline tests/test_ethereum_payment_provider.py]: I guess this is why the DAI/ETH order mattered above? I'm not sure that ordering should be tested here at all, though.

I was expecting the first option under the "Ethereum" payment provider to be ETH. I don't know how that gets decided. In the absence of specific direction, I have a slight preference for ETH-first.

Comment (pipermerriam) [reply tests/test_ethereum_payment_provider.py]: Yeah, I chose alphabetical "just because".  I'll update the test so that it is order independent.

File: tests/test_ethereum_payment_provider.py (hunk 3 above)
Comment (pipermerriam) [reply tests/test_ethereum_payment_provider.py]: payment provider.  I'll get the class name updated to something less utterly generic.

Comment (carver) [reply tests/test_ethereum_payment_provider.py]: Cool. So this text (`verbose_name` and `public_name`) is the text header where you choose between Bitcoin/CC/Ethereum payments? I guess `Ethereum-based` is the most accurate, since it includes DAI. No strong opinions though.

File: pretix_eth/payment.py (hunk 1 above)
Comment (davesque) [reply pretix_eth/payment.py]: Yeap, that's what order represents there.  And maybe just to keep things alphabetical in the form?
